src/core/secondaryAdapters/officialOnyxiaApiClient.ts

```diff
@@ -298,23 +298,39 @@ export function createOfficialOnyxiaApiClient(params: {
             () =>
                 axiosInstance
                     .get<{
-                        catalogs: {
-                            id: string;
-                            name: LocalizedString;
-                            location: string;
-                            description: LocalizedString;
-                            status: "PROD" | "TEST";
-                            catalog: {
-                                packages: {
-                                    description: string;
-                                    icon?: string;
-                                    name: string;
-                                    home?: string;
-                                }[];
-                            };
-                        }[];
-                    }>("/public/catalog")
-                    .then(({ data }) => data.catalogs),
+                        catalogs: Catalog[];
+                    }>(restApiPaths.catalogs)
+                    .then(({ data }) => {
+                        for (let catalog of data.catalogs) {
+                            let entries: Entry[] = [];
+                            for (let key of Object.keys(catalog.catalog.entries)) {
+                                if (key !== "library-chart") {
+                                    let entry = {
+                                        name: key,
+                                        values: catalog.catalog.entries[
+                                            key as keyof typeof catalog.catalog.entries
+                                        ],
+                                    } as unknown as Entry;
+                                    entry.latest = () =>
+                                        [...entry.values]
+                                            .sort((a, b) =>
+                                                a.version.localeCompare(
+                                                    b.version,
+                                                    undefined,
+                                                    {
+                                                        numeric: true,
+                                                        sensitivity: "base",
+                                                    },
+                                                ),
+                                            )
+                                            .pop()!;
+                                    entries.push(entry);
+                                }
+                            }
+                            catalog.catalog.entries = entries;
+                        }
+                        return data.catalogs;
```

Comment:
Can you comment, what is acheived here?

Reply:
The idea is to get an input catalog that looks like a chart helm, like [this](https://charts.bitnami.com/bitnami), as returned now by the API.
The problem is that the keys of the dictionary are not known in advance, hence the cast in Entry. I'm not sure if this is very clean but I don't know how to do it otherwise.

The "latest" function allows you to easily retrieve the latest Entry based on the version numbers.